File: Backend/Api/Models/Person.cs
using Api.Data.Models;

namespace Api.Models;

public class Person : PersonBase
{

    public long Id { get; set; }
}

public class PersonCreation : PersonBase { }

public abstract class PersonBase
{

    public string? Name { get; set; }

    public string? Surname { get; set; }

    public string? Identifier { get; set; }

    public DateTime? Birth { get; set; }

    public string? UserName { get; set; }

    public string? Password { get; set; }

    public UserType Type { get; set; }

    public string? Email { get; set; }

    public string? Phone { get; set; }

    public List<Right> Rights { get; set; } = new List<Right>();
}

File: flutter_frontend/lib/services/local/database/models/person.dart
import 'package:drift/drift.dart';
import 'package:helse/services/local/database/models/helse_table.dart';

class Person extends HelseTable {
  TextColumn get identifier => text().unique()();
  TextColumn get name => text().nullable()();
  TextColumn get types => text().nullable()();
  TextColumn get surname => text().nullable()();
  DateTimeColumn get birth => dateTime().nullable()();
  TextColumn get profilePicture => text().nullable()();
  TextColumn get email => text().nullable()();
  TextColumn get phone => text().nullable()();
}
